### PR TITLE
asio: update to 1.22.1

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update to @1.22.1.
Added a patch to fix an error caused by a missing examples folder (Homebrew did the same).
Added `muniversal` PortGroup to enable `+universal` build (tested on Leopard).

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

macOS 10.6 PPC (10A190)
Xcode 3.2

macOS 10.5.8
Xcode 3.1.4

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/asio/Portfile [hunk 1]
```diff
@@ -2,26 +2,30 @@
...
 name                        asio
-version                     1.20.0
+version                     1.22.1
 revision                    0
-checksums                   rmd160  762d0c9dec18b7ce0fdf9069067900ab1345887f \
-                            sha256  204374d3cadff1b57a63f4c343cbadcee28374c072dc04b549d772dbba9f650c \
-                            size    1845826
+checksums           rmd160  02ed50e944d1ebe835fa15d0e90dfd46c974ecf5 \
+                    sha256  6874d81a863d800ee53456b1cafcdd1abf38bbbf54ecf295056b053c0d7115ce \
+                    size    3085878
```
Comment (reneeotten) [inline devel/asio/Portfile:12]: please keep the same alignment

File: devel/asio/Portfile [hunk 2]
```diff
@@ -2,26 +2,30 @@
...
 
 license                     Boost-1
 description                 Asio C++ Library.
 long_description            Asio is a cross-platform C++ library for network and low-level I/O \
-                            programming that provides developers with a \
-                            consistent asynchronous model using a modern C++ approach.
+                            programming that provides developers \
+                            with a consistent asynchronous model using a modern C++ approach.
```
Comment (reneeotten) [inline devel/asio/Portfile:18]: please do not make unneeded whitespace changes, they clutter the diffs

File: devel/asio/Portfile [hunk 3]
```diff
@@ -2,26 +2,30 @@
...
 maintainers                 {gmail.com:g.litenstein @Lord-Kamina} openmaintainer
 categories                  devel
 
 homepage                    https://think-async.com/Asio/
 master_sites                sourceforge:project/${name}/${name}/${version}%20%28Stable%29
 use_bzip2                   yes
 
+# Tarball is missing `src/examples/cpp20`, which causes error
```
Comment (reneeotten) [inline devel/asio/Portfile:26]: did you report this to upstream?

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks @barracuda156 - a few comments

File: devel/asio/Portfile (hunk 2 above)
Comment (barracuda156) [reply devel/asio/Portfile:18]: These small things will never get fixed if no one ever bothers to do a one-time change.

File: devel/asio/Portfile (hunk 3 above)
Comment (barracuda156) [reply devel/asio/Portfile:26]: I’d be surprised if that has not been done already (I am not the first to discover the problem), but I can check it.

File: devel/asio/Portfile (hunk 2 above)
Comment (reneeotten) [reply devel/asio/Portfile:18]: I fail to see what problem you are "fixing" here

Comment (barracuda156) [reply devel/asio/Portfile:18]: It is not a compilation problem of course, but a bad typography. (Well, of course if there are good reasons for having it this way, I won’t insist.)

File: devel/asio/Portfile (hunk 1 above)
Comment (barracuda156) [reply devel/asio/Portfile:12]: > please keep the same alignment

Ok, will reverse that now.

File: devel/asio/Portfile (hunk 3 above)
Comment (barracuda156) [reply devel/asio/Portfile:26]: @reneeotten Let me actually move to Github from Sourceforce. The former has a tarball without examples missing: https://github.com/chriskohlhoff/asio/releases/tag/asio-1-22-1

**Comment by barracuda156** [comment]
@reneeotten Any idea why it fails on bots?
```
Executing:  cd "/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_asio/asio/work/asio-1-22-1/asio" && ./autogen.sh --verbose 
[294](https://github.com/macports/macports-ports/runs/6433712766?check_suite_focus=true#step:8:294)
  SDKROOT='/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk'
[295](https://github.com/macports/macports-ports/runs/6433712766?check_suite_focus=true#step:8:295)
  DEBUG: system:  cd "/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_asio/asio/work/asio-1-22-1/asio" && ./autogen.sh --verbose 
[296](https://github.com/macports/macports-ports/runs/6433712766?check_suite_focus=true#step:8:296)
  You need GNU autoconf to install from CVS (ftp://ftp.gnu.org/gnu/autoconf/)
[297](https://github.com/macports/macports-ports/runs/6433712766?check_suite_focus=true#step:8:297)
  Command failed:  cd "/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_asio/asio/work/asio-1-22-1/asio" && ./autogen.sh --verbose 
[298](https://github.com/macports/macports-ports/runs/6433712766?check_suite_focus=true#step:8:298)
  Exit code: 1
[299](https://github.com/macports/macports-ports/runs/6433712766?check_suite_focus=true#step:8:299)
  Error: Failed to configure asio: autoconf failure: command execution failed
[300](https://github.com/macports/macports-ports/runs/6433712766?check_suite_focus=true#step:8:300)
  DEBUG: Error code: NONE
[301](https://github.com/macports/macports-ports/runs/6433712766?check_suite_focus=true#step:8:301)
  DEBUG: Backtrace: autoconf failure: command execution failed
[302](https://github.com/macports/macports-ports/runs/6433712766?check_suite_focus=true#step:8:302)
      while executing
[303](https://github.com/macports/macports-ports/runs/6433712766?check_suite_focus=true#step:8:303)
  "$procedure $targetname"
```
Works for me locally.

Sourceforge version uses `configure` and builds everywhere, but requires a patch due to missing examples. Github version does not need a patch, but uses autotools, which for some reason fails on buildbots.

**Comment by barracuda156** [comment]
@reneeotten Should I rather just revert to Sourceforge version which builds fine across the board?

File: devel/asio/Portfile (hunk 2 above)
Comment (reneeotten) [reply devel/asio/Portfile:19]: I don't see anything wrong with it, both for content nor style - please leave as-is.

**Comment by reneeotten** [comment]
> @reneeotten Should I rather just revert to Sourceforge version which builds fine across the board?

I think the build dependencies need to coma *after* the "use_autoconf" statements

File: devel/asio/Portfile [hunk 4]
```diff
@@ -2,25 +2,34 @@
 
 PortSystem                  1.0
 PortGroup                   openssl 1.0
+PortGroup                   muniversal 1.0
+PortGroup                   github 1.0
 
-name                        asio
-version                     1.20.0
+github.setup                chriskohlhoff asio 1-22-1 asio-
+version                     1.22.1
```
Comment (reneeotten) [inline devel/asio/Portfile:9]: this should be done automatically by replacing the -" with "." using "${github.version}"

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: devel/asio/Portfile (hunk 2 above)
Comment (reneeotten) [reply devel/asio/Portfile:19]: @barracuda156 please revert this

Comment (barracuda156) [reply devel/asio/Portfile:19]: > I don't see anything wrong with it, both for content nor style - please leave as-is.

How about if I make this correction in a separate commit? (It _is_ wrong – articles and prepositions should not be “hanging” in the end of a line. I worked 10+ years in desktop publishing in academia, I know what I am talking about here.)

**Comment by reneeotten** [comment]
@barracuda156 I still think changing the line-continuations are useless with/without a separate commit but I've told you that several times now but you keep insisting on it... Whatever, it's fine with me. 

However please change your commit message and remove the `@` that has nothing to do with the version and also since you know change to the `github` PG you shouldn't need to muck around with the `livecheck.regex` anymore. So can you remove that line and see if the livecheck works correctly?

**Comment by barracuda156** [comment]
@reneeotten Yes, definitely. Sorry about @, I don’t use it in new PRs, but forgot to remove from earlier ones.

**Comment by mascguy** [comment]
Is this PR ready for merge?

**Comment by barracuda156** [comment]
> Is this PR ready for merge?

@mascguy I hope so :)

**Pull request merged**